### ISSUE TITLE
NVDA traps keyboard navigation and does not properly read aria list items in the group role.

### ISSUE THREAD

**Issue by delmyers** [nvaccess/nvda#7293]
## Steps to reproduce:

1. Save the html code listed below to a file
2. Start NVDA (version 2017.2)
3. Load the file in Internet Explorer or Microsoft Edge, or Google Chrome
4. Tab through the page until the outer box turns blue
5. Use left and right arrows.

## Expected Behavior
The circles should become "selected" by turning red, and NVDA should read their aria-labels.

## Actual Behavior
NVDA reads the aria labels one character at a time, and the circles do not become red.

Note, the behavior works as expected using Windows Narrator, and the "selection" behavior works correctly if NVDA is not started.

## Remarks

What I'm trying to do is make an accessible page that has a dynamic group if items that can be selected and interacted with (similar to what this page does without nvda turned on). I've tried other methods including aria grids, but grids do not seem to work for dynamic pages. I've found that if I add a grid cell to a row after the grid has been created, then the page crashes.

If someone could come up with an alternative for my needs that will work with NVDA, I would be happy to use that as well.

## Code

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <title>My Page</title>
    <style>
        h1 {
            color: red;
        }

        div.container {
            color: black;
            border-style: solid;
            border-width: 2px;
            border-color: black;
            width: 200px;
            height: 20px;
        }

        div.block
        {
            display:block;
        }

        div.dot {
            border-radius:50%;
            border-style:solid;
            border-width:2px;
            width:15px;
            height:15px;
            vertical-align:top;
            padding: 0px 0px 0px 0px;
            margin-left: 10px;
            float:left;
        }

        div.unselected {
            color: black;
        }

        div.selected {
            color:red;
        }
    </style>
    <script type="text/javascript">
        var selected = 0;

        function key(event) {
            var id = selected;

            if (event.keyCode == 39) {
                // right arrow
                id = id + 1;
                
            } else if (event.keyCode == 37) {
                // left arrow
                id = id - 1;
            } else {
                return;
            }

            select(id);

        }

        function select(id) {
            if (id) {

                var firstElement = document.getElementById(selected);
                var element = document.getElementById(id);
                if (element) {
                    if (firstElement) {
                        firstElement.className = "dot unselected";
                        element.ariaSelected = false;
                    }
                    selected = id;
                    element.focus();
                    element.className = "dot selected";
                    element.ariaSelected = true;
                    //event.which.className = "dot";
                }
            }
        }

        function selectpanel()
        {
            var panel = document.getElementById("panel");
            if (panel) {
                panel.style.borderColor = "blue";
                panel.style.borderWidth = "4px";
            }
        }

        function unselectpanel() {
            var panel = document.getElementById("panel");
            if (panel) {
                panel.style.borderColor = "black";
                panel.style.borderWidth = "2px";
            }
        }

    </script>
</head>
<body>
    <h1>Here is a page</h1>
    <div aria-label="outer container" class="container" tabindex=0 id="panel" onkeyup="key(event)" onfocus="selectpanel()" onblur="unselectpanel()">
        <div class="block"  tablindex="-1" role="group" tabindex=-1 id="-1">
            <div class="dot" role="listitem" id="1" aria-label="dot one" title="dot one" onmouseup="select(1)"> </div>
            <div class="dot" role="listitem" id="2" aria-label="dot two" title="dot two" onmouseup="select(2)"> </div>
            <div class="dot" role="listitem" id="3" aria-label="dot three" title="dot three" onmouseup="select(3)"> </div>
            <div class="dot" role="listitem" id="4" aria-label="dot four" title="dot four" onmouseup="select(4)"> </div>
            <div class="dot" role="listitem" id="5" aria-label="dot five" title="dot five"onmouseup="select(5)" > </div>
        </div>
     </div>

</body>
</html>
```

**Comment by josephsl** [comment]
Hi, okay, what about testing this with Firefox? Thanks. CC @DerekRiemer

**Comment by jcsteh** [comment]
You noted that this works with Narrator, but I can't get this to work. Left and right arrows do not read the labels. I see your code tries to focus the lits items, but the list items do not have a tabindex, so this won't work.

Regardless, the reason that NVDA overrides the arrow keys here is that listitem and group are not interactive widgets, so NVDA uses browse mode, where the arrow keys among other things are instead used to navigate a document like you would in a word processor. NVDA automatically switches to focus mode (where these keys are all passed through to the browser) for interactive widget roles. An ARIA grid would certainly achieve this. If you're seeing a browser crash with grids, that suggests something needs to be fixed somewhere, perhaps in the browser.

In any case, NVDA is behaving as expected here, since these aren't interactive roles.
